scrape/scrape.go

```diff
@@ -574,20 +573,16 @@ func (sp *scrapePool) sync(targets []*Target) {
 // refreshTargetLimitErr returns an error that can be passed to the scrape loops
 // if the number of targets exceeds the configured limit.
 func (sp *scrapePool) refreshTargetLimitErr() error {
-	if sp.config == nil || sp.config.TargetLimit == 0 && !sp.targetLimitHit {
+	if sp.config == nil || sp.config.TargetLimit == 0 {
```

Comment:
```suggestion
	if sp.config == nil || sp.config.TargetLimit == 0 || !sp.targetLimitHit {
```

That seems to be a fix that would keep the short path